### PR TITLE
Replaced java.xml.bind with jakarta.xml.bind, Fixes https://github.com/SAP/scimono/issues/153

### PR DESCRIPTION
**Pull request by SherwinVarghese**
- Replaced `java.xml.bind` with `jakarta.xml.bind`
- Fixes build issues in Java 11
- Fixes https://github.com/SAP/scimono/issues/153

Reach out to `sherwin.varghese@sap.com` if needed.

### CODE REVIEW

File: scimono-client/pom.xml [hunk 1]
```diff
@@ -85,5 +85,16 @@
...
+          <groupId>jakarta.xml.bind</groupId>
+          <artifactId>jakarta.xml.bind-api</artifactId>
+          <version>3.0.0</version>
+        </dependency>
+        <dependency>
+          <groupId>com.sun.xml.bind</groupId>
+          <artifactId>jaxb-impl</artifactId>
+          <version>3.0.0</version>
+          <scope>runtime</scope>
+        </dependency>
```
Comment (karaimin) [inline scimono-client/pom.xml:98]: Can you place this in the existing `<dependencyManagement>` configuration (in the parent pom) and remove the `<version>` attribute here ?

Once, you move the definition to  `<dependencyManagement>`, please extract the version as a separate maven variable (like others).

Also, there is newer version: 3.0.1. Can you use it instead?

**Review by karaimin — CHANGES_REQUESTED** [review]
Thank you for your contribution. I would like to ask you to change a few things :)

File: scimono-client/pom.xml (hunk 1 above)
Comment (SherwinVarghese) [reply scimono-client/pom.xml:106]: done @karaimin.

The above dependencies passed the build however there were Class Not Found exceptions while running the JAR, so the below dependencies were added instead - 

```
<dependency>
      <groupId>javax.xml.bind</groupId>
      <artifactId>jaxb-api</artifactId>
</dependency>
<dependency>
      <groupId>org.glassfish.jaxb</groupId>
      <artifactId>jaxb-runtime</artifactId>
</dependency>  
```

Comment (SherwinVarghese) [reply scimono-client/pom.xml:106]: tested the `scimono-compliance-tests-0.0.27-SNAPSHOT-shaded.jar` with the current dependencies and it is working.

Comment (karaimin) [reply scimono-client/pom.xml:106]: Hi @SherwinVarghese,
I don't think that you should add these new dependencies. Actually, if you look at the StackOverflow answer (provided by you in the description), it is described that `jakarta` should be enough (no need of the old ones). 

I suppose the reason that you have runtime issues is that you declare the implementation dependency with runtime `scope`, and no classes exist in the backend. 

```xml
<dependency>
          <groupId>com.sun.xml.bind</groupId>
          <artifactId>jaxb-impl</artifactId>
          <version>3.0.0</version>
          <scope>runtime</scope>
</dependency>
```
Can you  remove the `runtime` scope and try again?

Comment (SherwinVarghese) [reply scimono-client/pom.xml:106]: Hi @karaimin,

 Even after removing the `runtime` scope, the build passes but there are runtime exceptions.
 The root cause is `jersey-bom` is on an older version that depends on `javax.xml.bind`.

While updating the `jersey-bom` to `3.0.1`, there are jakarta exceptions.


```
[ERROR] testParsingInvalidFilterSyntax{String}[29]  Time elapsed: 0.003 s  <<< FAILURE!
org.opentest4j.AssertionFailedError: Unexpected exception type thrown ==> expected: <com.sap.scimono.exception.InvalidFilterException> but was: <java.lang.NoClassDefFoundError>
	at com.sap.scimono.fiter.QueryFilterParserTest.testParsingInvalidFilterSyntax(QueryFilterParserTest.java:89)
Caused by: java.lang.NoClassDefFoundError: jakarta/ws/rs/ext/RuntimeDelegate
```

This seems to  be the best fix that executes well during the runtime.

**Comment by karaimin** [comment]
Fixed in #160

**Pull request closed without merging**